### PR TITLE
App-sre migration from saas-analytics

### PR DESCRIPTION
**Pull request by deepak1725**
# Description

App-sre migration from saas-analytics

Fixes # (issue)
https://issues.redhat.com/browse/APPAI-1810

Equivalent PR in Worker: https://github.com/fabric8-analytics/fabric8-analytics-worker/pull/950 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Dockerfile.app-sre [hunk 1]
```diff
@@ -0,0 +1,24 @@
+FROM quay.io/app-sre/fabric8-analytics-f8a-license-analysis
```
Comment (prashbnair) [inline Dockerfile.app-sre:1]: Can you not use ubi images?

File: Dockerfile.app-sre [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN yum install -y epel-release && \
+    yum install -y python36-setuptools git gcc gcc-c++ python36-pip python36-devel \
+    libffi-devel openssl-devel && yum clean all
+
+
+# install python packages
+COPY ./requirements.txt /
+RUN pip3 install -r requirements.txt && rm requirements.txt
+RUN pip3 install git+https://github.com/fabric8-analytics/fabric8-analytics-auth.git@${F8A_AUTH_VERSION}
```
Comment (prashbnair) [inline Dockerfile.app-sre:16]: is this not handled in requirements.txt

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: Dockerfile.app-sre (hunk 2 above)
Comment (deepak1725) [reply Dockerfile.app-sre:16]: This is an exact copy of the existing Docker file, did modifications in base image only

File: Dockerfile.app-sre (hunk 1 above)
Comment (deepak1725) [reply Dockerfile.app-sre:1]: i thought image names should match with what we gave in SRE, https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/15370/diffs

File: build_deploy.rhel.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# login to quay to pull base image
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
+
+# build the image
+docker build  --no-cache \
+              --force-rm \
+              -t ${IMG}  \
+              -f ./Dockerfile .
```
Comment (sawood14012) [inline build_deploy.rhel.sh:19]: this should build your rhel docker file that is `Dockerfile.rhel.app-sre`

File: build_deploy.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+QUAY_IMAGE="quay.io/app-sre/${BASE_IMG}"
+IMG="${BASE_IMG}:latest"
+
+GIT_HASH=`git rev-parse --short=7 HEAD`
+
+# build the image
+docker build  --no-cache \
+              --force-rm \
+              -t ${IMG}  \
+              -f ./Dockerfile .
```
Comment (sawood14012) [inline build_deploy.sh:14]: similar way here `Dockerfile.app-sre`

File: Dockerfile.rhel.app-sre [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+RUN yum install -y epel-release && \
+    yum install -y python36-setuptools git gcc gcc-c++ python36-pip python36-devel \
+    libffi-devel openssl-devel && yum clean all
+
+
+# install python packages
+COPY ./requirements.txt /
+RUN pip3 install -r requirements.txt && rm requirements.txt
+RUN pip3 install git+https://github.com/fabric8-analytics/fabric8-analytics-auth.git@${F8A_AUTH_VERSION}
```
Comment (sawood14012) [inline Dockerfile.rhel.app-sre:14]: please include auth in` requirements.txt `

**Review by sawood14012 — CHANGES_REQUESTED** [review]

File: Dockerfile.app-sre (hunk 1 above)
Comment (prashbnair) [reply Dockerfile.app-sre:1]: You should move to UBI images and get rid of the RHEL build.
https://github.com/fabric8-analytics/f8a-stacks-report/blob/master/Dockerfile.new

File: build_deploy.rhel.sh (hunk 3 above)
Comment (deepak1725) [reply build_deploy.rhel.sh]: Ack

File: build_deploy.sh (hunk 4 above)
Comment (deepak1725) [reply build_deploy.sh]: Ack.

File: Dockerfile.app-sre (hunk 1 above)
Comment (deepak1725) [reply Dockerfile.app-sre]: Ack

File: Dockerfile.new [hunk 6]
```diff
@@ -0,0 +1,24 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+ENV F8A_AUTH_VERSION=fff8f49
+
+
+ENV F8A_AUTH_VERSION=fff8f49
```
Comment (prashbnair) [inline Dockerfile.new:6]: Line repeated

File: Dockerfile.new [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN yum install -y epel-release && \
+    yum install -y python36-setuptools git gcc gcc-c++ python36-pip python36-devel \
+    libffi-devel openssl-devel && yum clean all
+
+
+# install python packages
+COPY ./requirements.txt /
+RUN pip3 install -r requirements.txt && rm requirements.txt
```
Comment (prashbnair) [inline Dockerfile.new:15]: --no-cache is required to reduce image size.

File: Dockerfile.new [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+RUN pip3 install git+https://github.com/fabric8-analytics/fabric8-analytics-auth.git@${F8A_AUTH_VERSION}
```
Comment (prashbnair) [inline Dockerfile.new:16]: Can you move this to requirements.in?

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: Dockerfile.new (hunk 6 above)
Comment (deepak1725) [reply Dockerfile.new]: removed

File: Dockerfile.new (hunk 7 above)
Comment (deepak1725) [reply Dockerfile.new]: Ack.

File: Dockerfile.new (hunk 8 above)
Comment (deepak1725) [reply Dockerfile.new]: done

File: Dockerfile.new [hunk 9]
```diff
@@ -0,0 +1,18 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+RUN yum install -y epel-release && \
```
Comment (prashbnair) [inline Dockerfile.new:3]: Have you tested this, I dont think ubi-minimal has yum, you have to use microdnf.

File: Dockerfile.new [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    yum install -y python36-setuptools git gcc gcc-c++ python36-pip python36-devel \
+    libffi-devel openssl-devel && yum clean all
+
+
+# install python packages
+COPY ./requirements.txt /
+RUN pip3 install -r requirements.txt && rm requirements.txt --no-cache
```
Comment (prashbnair) [inline Dockerfile.new:10]: Can you also check if it --no-cache or --no-cache-dir?

File: Dockerfile.rhel.app-sre (hunk 5 above)
Comment (deepak1725) [reply Dockerfile.rhel.app-sre]: Ack

File: Dockerfile.new (hunk 9 above)
Comment (deepak1725) [reply Dockerfile.new]: done, Updated image with ubi-minimal

File: Dockerfile.new (hunk 10 above)
Comment (deepak1725) [reply Dockerfile.new]: yup, its `--no-cache-dir` Updated

**Review by prashbnair — APPROVED** [review]

**Review by sawood14012 — APPROVED** [review]

**Pull request merged**